code/game/machinery/computer/card.dm

```diff
@@ -648,95 +793,104 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
-			switch(account.account_department) //Checking if the user has access to change pay.
-				if(ACCOUNT_SRV,ACCOUNT_CIV,ACCOUNT_CAR)
-					if(!(ACCESS_HOP in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-				if(ACCOUNT_SEC)
-					if(!(ACCESS_HOS in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-				if(ACCOUNT_MED)
-					if(!(ACCESS_CMO in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-				if(ACCOUNT_SCI)
-					if(!(ACCESS_RD in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-				if(ACCOUNT_ENG)
-					if(!(ACCESS_CE in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-			var/new_pay = FLOOR(input(usr, "Input the new paycheck amount.", "Set new paycheck amount.", account.paycheck_amount) as num|null, 1)
+			if(SSeconomy.is_nonstation_account(B))
+				CRASH("[ckey(src)] tried to adjust [B.account_id] payment. It must be they're hacking the game.")
+			var/new_pay = FLOOR(input(usr, "Input the new paycheck amount.", "Set new paycheck amount.", B.payment_per_department[target_paycheck]) as num|null, 1)
 			if(isnull(new_pay))
 				updateUsrDialog()
 				return
 			if(new_pay < 0)
 				to_chat(usr, "<span class='warning'>Paychecks cannot be negative.</span>")
 				updateUsrDialog()
 				return
-			account.paycheck_amount = new_pay
+			B.payment_per_department[paycheck_t] = new_pay
 
 		if ("adjust_bonus")
 			//Adjust the bonus pay of a crew member. Negative amounts dock pay.
-			if(!inserted_scan_id)
+			if(!authenticated)
 				updateUsrDialog()
 				return
-			var/account_name = href_list["account"]
-			var/datum/bank_account/account = null
-			for(var/datum/bank_account/B in SSeconomy.bank_accounts)
-				if(B.account_holder == account_name)
-					account = B
-					break
-			if(isnull(account))
+			var/paycheck_t = href_list["paycheck_t"]
+			var/datum/bank_account/B = inserted_modify_id?.registered_account
+			if(isnull(B))
 				updateUsrDialog()
 				return
-			switch(account.account_department) //Checking if the user has access to change pay.
-				if(ACCOUNT_SRV,ACCOUNT_CIV,ACCOUNT_CAR)
-					if(!(ACCESS_HOP in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-				if(ACCOUNT_SEC)
-					if(!(ACCESS_HOS in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-				if(ACCOUNT_MED)
-					if(!(ACCESS_CMO in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-				if(ACCOUNT_SCI)
-					if(!(ACCESS_RD in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-				if(ACCOUNT_ENG)
-					if(!(ACCESS_CE in inserted_scan_id.access))
-						updateUsrDialog()
-						return
-			var/new_bonus = FLOOR(input(usr, "Input the bonus amount. Negative values will dock paychecks.", "Set paycheck bonus", account.paycheck_bonus) as num|null, 1)
+			if(SSeconomy.is_nonstation_account(B))
+				CRASH("[ckey(src)] tried to adjust [inserted_modify_id.registered_name]'s [B.account_id] pay bonus. It must be they're hacking the game.")
```

Comment:
key_name instead of ckey. Also use message_admins, log_admin_private, and return instead of CRASH

Reply:
message_admins should use ADMIN_LOOKUPFLW or whatever it is